src/util/autoclose.ts

```diff
@@ -10,24 +10,22 @@ const isContainedIn = (element: HTMLElement, array?: HTMLElement[]) =>
 const matchesSelectorIfAny = (element: HTMLElement, selector?: string) =>
     !selector || closest(element, selector) != null;
 
-// we'll have to use 'touch' events instead of 'mouse' events on iOS and add a more significant delay
-// to avoid re-opening when handling (click) on a toggling element
+// we have to add a more significant delay to avoid re-opening when handling (click) on a toggling element
 // TODO: use proper Angular platform detection when NgbAutoClose becomes a service and we can inject PLATFORM_ID
-let iOS = false;
-if (typeof navigator !== 'undefined') {
-  iOS = !!navigator.userAgent && /iPad|iPhone|iPod/.test(navigator.userAgent);
-}
+const isMobile = () => typeof navigator !== 'undefined' ?
+    !!navigator.userAgent && /iPad|iPhone|iPod|Android/.test(navigator.userAgent) :
```

Comment:
Unfortunately this won't work for iPads with iOS >= 13. Since this version `userAgent` does not contain `iPad`.

https://stackoverflow.com/questions/2153877/what-is-the-ipad-user-agent/56923008#56923008

I checked it on browserstack and I can confirm that the issue still persists. This commit did the fix for Android though

Reply:
Yikes, thanks for the info. Of course we should not use `userAgent` at all for this...